src/internal/util/errors.ts

```diff
@@ -0,0 +1,68 @@
+
+/**
+ * Used to identify different types of RxJS errors.
+ */
+export const enum RxErrorCode {
+  Empty = 0,
+  OutOfRange = 1,
+  ObjectUnsubscribed = 2,
+  Timeout = 3,
+  Teardown = 4,
+}
+
+/**
+ * Creates an error and decorates it with the appropriate error code for identification later.
+ */
+export function createRxError(message: string, code: RxErrorCode, ErrorType: any = Error) {
+  const result = new ErrorType('RxJS: ' + message);
+  (result as any).__rxjsErrorCode = code;
```

Comment:
Would there be any advantage in using a symbol instead of a string key for the `__rxjsErrorCode` property?

Reply:
There would be one disadvantage, which is it wouldn't work in IE. that's really it.